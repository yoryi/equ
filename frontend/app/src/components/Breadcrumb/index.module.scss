:local(.container) {
    margin: 0 -8px;

    a {
        color: var(--main-dark);

        &:hover {
            text-decoration: underline;
        }
    }

    span {
        margin: 0 8px;
        font-size: 12px;
        line-height: 18px;
        color: var(--main-dark);

        &:global(.active) {
            color: var(--grey);
        }
    }
}
