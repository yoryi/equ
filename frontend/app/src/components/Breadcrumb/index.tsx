//Utils
import cx from "classnames"
import * as React from "react"

//Icons
import { ReactComponent as ArrowRight } from "../../assets/arrow-right.svg"
//Styles
import Styles from "./index.module.scss"

interface BreadcrumbProps {
  className?: string
  children: React.ReactChild[]
}

const Breadcrumb: React.VFC<BreadcrumbProps> = ({ className, children }) => (
  <div className={cx(Styles.container, className)}>
    {children.reduce<React.ReactChild[]>((elements, child, i) => {
      if (i === elements.length - 1) {
        return [...elements, child]
      }

      return elements.concat(child, <ArrowRight key={`arrow-${i}`} />)
    }, [])}
  </div>
)

export default Breadcrumb
