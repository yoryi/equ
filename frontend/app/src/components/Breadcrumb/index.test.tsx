//Testing
import { render } from "@testing-library/react"
import faker from "faker"
import _ from "lodash"
import * as React from "react"
//Navigation
import { Link,MemoryRouter } from "react-router-dom"

//Components
import Breadcrumb from "./"

describe(`Breadcrumb component`, () => {
  faker.seed(1000)

  const wrapper: React.FC = ({ children }) => (
    <MemoryRouter>{children}</MemoryRouter>
  )

  it(`should render and match the snapshot`, () => {
    const { container } = render(
      <Breadcrumb>
        <Link to={`/${faker.lorem.word()}`}>{faker.lorem.word()}</Link>

        <span className={`active`}>{faker.lorem.word()}</span>
      </Breadcrumb>,
      { wrapper },
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
