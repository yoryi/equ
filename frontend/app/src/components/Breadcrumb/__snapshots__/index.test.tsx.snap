// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component should render and match the snapshot 1`] = `
<div
  class="container"
>
  <a
    href="/et"
  >
    quia
  </a>
  <svg>
    arrow-right.svg
  </svg>
  <span
    class="active"
  >
    fugit
  </span>
</div>
`;
